buildpack/files.go

```diff
@@ -192,9 +197,57 @@ func containsName(unmet []Unmet, name string) bool {
 
 // layer content metadata
 
-type LayerMetadataFile struct {
-	Data   interface{} `json:"data" toml:"metadata"`
-	Build  bool        `json:"build" toml:"build"`
-	Launch bool        `json:"launch" toml:"launch"`
-	Cache  bool        `json:"cache" toml:"cache"`
+type EncoderDecoder interface {
+	IsSupported(buildpackAPI string) bool
+	Encode(file *os.File, lmf layertypes.LayerMetadataFile) error
+	Decode(path string) (layertypes.LayerMetadataFile, string, error)
+}
+
+func EncodeFalseFlags(lmf layertypes.LayerMetadataFile, path, buildpackAPI string) error {
+	fh, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer fh.Close()
+
+	lmf.UnsetFlags()
+
+	encoders := []EncoderDecoder{
```

Comment:
nit: something like `DefaultEncoders()` would allow us to add/remove a version encoder in one spot in the future